protocol/contracts/templegold/SpiceAuction.sol

```diff
@@ -0,0 +1,494 @@
+        if (epochInfo.startTime == 0) { revert InvalidEpoch(); }
+        if (!epochInfo.hasEnded()) { revert AuctionActive(); }
+
+        SpiceAuctionConfig storage _config = auctionConfigs[epochId];
+        (address bidToken,) = _getBidAndAuctionTokens(_config);
+        if (bidToken != templeGold) { revert CommonEventsAndErrors.InvalidParam(); }
+        uint256 amount = epochInfo.totalBidTokenAmount;
+        if (amount == 0) { revert CommonEventsAndErrors.ExpectedNonZero(); }
+        
+        emit RedeemedTempleGoldBurned(epochId, amount);
+        redeemedEpochs[epochId] = true;
+        _burnAndNotify(amount, _config.recipient, useContractEth);
+    }
+
+    /**
+     * @notice Get spice auction config for an auction
+     * @param auctionId Id of auction
+     */
+    function getAuctionConfig(uint256 auctionId) external view override returns (SpiceAuctionConfig memory) {
+        return auctionConfigs[auctionId];
+    }
+
+    /**
+     * @notice Get auction token for current epoch
+     * @return Auction token
+     */
+    function getAuctionTokenForCurrentEpoch() external override view returns (address) {
+        SpiceAuctionConfig memory config = auctionConfigs[_currentEpochId];
+        return config.isTempleGoldAuctionToken ? templeGold : spiceToken;
+    }
+
+    /**
+     * @notice Get current epoch
+     * @return Epoch Id
+     */
+    function currentEpoch() external view override returns (uint256) {
+        return _currentEpochId;
+    }
+
+    /**
+     * @notice Get claimable amount for an epoch
+     * @dev function will return claimable for epoch. This can change with more user deposits
+     * @param depositor Address to check amount for
+     * @param epochId Epoch id
+     * @return Claimable amount
+     */
+    function getClaimableForEpoch(address depositor, uint256 epochId) external override view returns (uint256) {
+        if (claimed[depositor][epochId]) { return 0; }
+        uint256 bidTokenAmount = depositors[depositor][epochId];
+        if (bidTokenAmount == 0 || epochId > _currentEpochId) { return 0; }
+        EpochInfo memory info = epochs[epochId];
+        return bidTokenAmount.mulDivRound(info.totalAuctionTokenAmount, info.totalBidTokenAmount, false);
+    }
+
+    function _getBidAndAuctionTokens(
+        SpiceAuctionConfig storage _config
+    ) private view returns (address bidToken, address auctionToken) {
+        (bidToken, auctionToken) = _config.isTempleGoldAuctionToken
+            ? (spiceToken, templeGold)
+            : (templeGold, spiceToken);
+    }
+
+    function _burnAndNotify(uint256 amount, address from, bool useContractEth) private {
+        // pull funds from bids recipient (set in config)
+        IERC20(templeGold).safeTransferFrom(from, address(this), amount);
+        // burn directly and call TempleGold to update circulating supply
+        if (block.chainid == _mintChainId) {
+            ITempleGold(templeGold).burn(amount);
+            return;
```

Comment:
When `block.chainid == _mintChain`,  any potential `msg.value` is not returned to the caller and is donated to the contract because of this `return` statement here. 

I know most users won't make such mistake, but just in case, I would enforce that if `block.chainid == _mintChainId`, then include `require(msg.value == , "ether not needed")` in the public `burnAndNotify()`. 

Luckily any donated eth can be recovered with the `withdrawEth()` function.

Reply:
Good, fixed